syntax = "proto3";
package pb;

import "google/protobuf/timestamp.proto";

service Feed {
    rpc GetFeed(GetFeedRequest) returns (GetFeedResponse) {}
}

message GetFeedRequest {
    string username = 1;
    string startToken = 2;
}

enum EventType {
    POST_ADDED = 0;
    POST_UPDATED = 1;
    POST_DELETED = 2;
}

message Event  {
	EventType eventType = 1;
	string username = 2;
	string url = 3;
	google.protobuf.Timestamp timestamp = 4;
}

message GetFeedResponse {
	repeated Event events = 1;
	string nextToken = 2;
    string err = 3;
}

